docs/book/reflection-abstract-factory.md

```diff
@@ -28,26 +58,30 @@ return [
 ];
 ```
 
-Mapping services to the factory is more explicit and performant.
 
-The factory operates with the following constraints/features:
+### Using Mezzio
 
-- A parameter named `$config` typehinted as an array will receive the
-  application "config" service (i.e., the merged configuration).
-- Parameters typehinted against array, but not named `$config`, will
-  be injected with an empty array.
-- Scalar parameters will result in the factory raising an exception,
-  unless a default value is present; if it is, that value will be used.
-- If a service cannot be found for a given typehint, the factory will
-  raise an exception detailing this.
+In a module's [ConfigProvider](https://docs.laminas.dev/laminas-config-aggregator/config-providers/) class:
 
-`$options` passed to the factory are ignored in all cases, as we cannot
-make assumptions about which argument(s) they might replace.
+- Add the `ReflectionBasedAbstractFactory` to the `abstract_factories` array returned from the `getDependencies()` method.
+- Use the `ReflectionBasedAbstractFactory` to instantiate the class, in the `factories` array returned from the `getDependencies()` method.
 
-Once your dependencies have stabilized, we recommend writing a dedicated
-factory, as reflection can introduce performance overhead; you may use the
-[generate-factory-for-class console tool](console-tools.md#generate-factory-for-class)
-to do so.
+```php
+use Laminas\ServiceManager\AbstractFactory\ReflectionBasedAbstractFactory;
+
+public function getDependencies(): array
+{
+    return [
+        'abstract_factories' => [
+            ReflectionBasedAbstractFactory::class,
```

Comment:
I think we should note that the `ReflectionBasedAbstractFactory` has security implications, if user input is passed to the service locator 🤔

Reply:
`$locator->get('SomeClassThatWillDoSomeInputOutputOnConstruct')` is a valid DDoS vector, for example

Reply:
Actually, @Ocramius, can you step me through the implications further, please?

Reply:
```php
class IWillDoDamage
{
    public function __construct() { /* runs `rm -rf /` here */ }
}

$serviceLocator->get($_POST['service']);
```

Then send a POST request with `service=IWillDoDamage`

Reply:
Curiously, why would you let someone pull a service that way? Sorry if I'm just missing the obvious.